internal/api/mail.go

```diff
@@ -650,21 +650,22 @@ func (a *API) sendEmail(r *http.Request, tx *storage.Connection, u *models.User,
 	}
 
 	if config.Hook.SendEmail.Enabled {
-		// When secure email change is disabled, we place the token for the new email on emailData.Token
-		if emailActionType == mail.EmailChangeVerification && !config.Mailer.SecureEmailChangeEnabled && u.GetEmail() != "" {
-			otp = otpNew
-		}
-
 		emailData := mail.EmailData{
 			Token:           otp,
```

Comment:
> Currently, there is no token passed in into the email hook when an email change is attempted right after a phone sign up.

what is the initial value of `otp` here?

Reply:
Should be `""` if there's secure email change is not enabled. If it's enabled, and there's no phone linked to it it doesn't get set so it remains as `""`. 

 [ref]( https://github.com/supabase/auth/blob/master/internal/api/mail.go#L513)